user-interface/src/main/java/life/qbic/datamanager/views/projects/project/samples/registration/batch/EditBatchDialog.java

```diff
@@ -0,0 +1,228 @@
+import life.qbic.projectmanagement.domain.model.batch.BatchId;
+import life.qbic.projectmanagement.domain.model.experiment.ExperimentalGroup;
+import life.qbic.projectmanagement.domain.model.experiment.vocabulary.Analyte;
+import life.qbic.projectmanagement.domain.model.experiment.vocabulary.Species;
+import life.qbic.projectmanagement.domain.model.experiment.vocabulary.Specimen;
+
+/**
+ * A dialog used for sample batch registration.
+ */
+public class EditBatchDialog extends DialogWindow {
+
+  private final SampleBatchInformationSpreadsheet spreadsheet;
+  private final BatchId batchId;
+  private final List<SampleInfo> existingSamples;
+
+  private final TextField batchNameField;
+  private final Span batchNameText;
+
+  public EditBatchDialog(String experimentName,
+      List<Species> species,
+      List<Specimen> specimen,
+      List<Analyte> analytes,
+      List<ExperimentalGroup> experimentalGroups,
+      BatchId batchId,
+      String batchName,
+      List<SampleInfo> existingSamples) {
+
+    addClassName("batch-update-dialog");
+    setConfirmButtonLabel("Update Samples");
+
+    this.batchId = batchId;
+    this.existingSamples = Collections.unmodifiableList(existingSamples);
+
+    spreadsheet = new SampleBatchInformationSpreadsheet(experimentalGroups, species, specimen,
+        analytes, false);
+
+    batchNameField = new TextField();
+    batchNameField.addClassName("batch-name-field");
+    batchNameField.setLabel("Batch Name");
+    batchNameField.setPlaceholder("Please enter a name for this batch");
+    batchNameField.setRequired(true);
+    // must contain at least one non-whitespace character and no leading/tailing whitespace.
+    batchNameField.setPattern("^\\S+(.*\\S)*$");
+    batchNameField.setErrorMessage(
+        "The batch name must not be empty. It must not start nor end with whitespace.");
+    batchNameField.setValue(batchName);
+    batchNameField.addValueChangeListener(this::onBatchNameChanged);
+
+    Button addRow = new Button();
+    addRow.setText("Add Row");
+    addRow.addClickListener(this::onAddRowClicked);
+    addRow.addClassName("add-batch-row");
+
+    Button removeLastRow = new Button();
+    removeLastRow.setText("Remove Row");
+    removeLastRow.addClickListener(this::onRemoveLastRowClicked);
+    removeLastRow.addClassName("remove-batch-row");
+
+    setHeaderTitle("Update Batch");
+    setResizable(true);
+
+    Div batchControls = new Div();
+    batchControls.addClassName("batch-controls");
+    batchControls.add(batchNameField);
+
+    Span userHelp1 = new Span("You are editing samples for batch");
+    Span experimentNameText = new Span(experimentName);
+    experimentNameText.addClassName("experiment-name");
+    Span userHelp2 = new Span("and experiment");
+    batchNameText = new Span(batchName);
+    batchNameText.addClassName("batch-name");
+    Div userHelpText = new Div(userHelp1, batchNameText, userHelp2, experimentNameText);
+    userHelpText.addClassName("user-help-text");
```

Comment:
This could also be provided by a seperate method 🤔

Reply:
I do think these belong in the same method. The structure of the dialog is described in this method (the constructor) and thus I favor cohesion here.